locustempus/main/views.py

```diff
@@ -38,61 +37,69 @@
 
 
 class IndexView(LoginRequiredMixin, View):
-    template_name = 'main/index.html'
+    template_name = 'main/course_list.html'
     http_method_names = ['get']
 
-    def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
-        # If user has only one course, redirect to course, else render a list
-        # of courses
+    def get(self, request, *args, **kwargs) -> HttpResponse:
         courses = get_courses_for_user(self.request.user)
-        n = courses.count()
-
-        if n == 0:
-            # Render a view with no courses
-            ctx = {
-                'user': request.user,
-                'courses': None
-            }
-            return render(request, self.template_name, ctx)
-        elif n == 1:
-            # Redirect to the course
-            course = courses.first()
-            return HttpResponseRedirect(
-                reverse('course-detail-view', kwargs={'pk': course.pk}))
-        else:
-            # Render a list of courses
-            return render(request, self.template_name, {
-                'user': request.user,
-                'courses': courses,
-            })
+        ctx = {
+            'user': request.user,
+            'registrar_courses': courses.filter(info__term__isnull=False),
+            'sandbox_courses': courses.filter(info__term__isnull=True)
+        }
+        return render(request, self.template_name, ctx)
 
 
-class CourseCreateView(LoggedInSuperuserMixin, CreateView):
+class CourseCreateView(LoginRequiredMixin, CreateView):
     model = Course
     template_name = 'main/course_create.html'
-    fields = ['title', 'group', 'faculty_group']
+    fields = ['title']
+
+    @staticmethod
+    def get_group_names(course_title: str) -> Tuple[str, str]:
```

Comment:
I see that you're trying to prevent duplication here. I'm wondering if tacking a uuid on the user generated course title might work better? This will also serve to match up the group names.

Reply:
sounds like a much better approach